h2o-core/src/main/java/hex/Model.java

```diff
@@ -663,6 +680,7 @@ protected Output(ModelBuilder b, Frame train) {
         throw new IllegalArgumentException(b.validationErrors());
       // Capture the data "shape" the model is valid on
       setNames(train != null ? train.names() : new String[0]);
+      setColumnTypes(train);
```

Comment:
this can cause NullPointerException - train can be null

Reply:
I added the null pointer check in the method setColumnTypes.  Sounds okay?

Reply:
I changed it to setColumnTypes(train!=null?train.typesStr():new String[0]).  This should fix it.